rc/tools/lint.kak

```diff
@@ -89,7 +89,16 @@ define-command \
                     '
                         BEGIN { OFS=":"; FS=":" }
 
-                        /:[1-9][0-9]*:[1-9][0-9]*:/ {
+                        # without column
+                        /^[^:]+:[1-9][0-9]*: / {
+                            $1 = ENVIRON["kak_bufname"]
+                            $2 += line_offset
+                            $2 = $2 FS "x";
+                            print $0
+                        }
+
+                        # with column
+                        /^[^:]+:[1-9][0-9]*:[1-9][0-9]*: / {
```

Comment:
could not capture groups do the trick to mostly share the code between with/without column ?

Reply:
I'm not too familiar with awk and it looked like working with capture groups is either non-trivial or non-portable.

Here's my attempt at making the code more DRY: b83b576028685b6d160f1fb2e7ca2aae5c0f5a6f

Does this work for you?